From 31bab06c8a3cc0318847018e245a91dedaf84d04 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 1 Jan 2026 04:14:14 +0100
Subject: [PATCH] csd-background: Remove decoration from desktop window

Without this, window decoration is drawn and the background is shifted
in XWayland if `GTK_CSD=1` environment variable is set.
---
 plugins/background/monitor-background.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/background/monitor-background.c b/plugins/background/monitor-background.c
index 2cbfcb2..ba0853e 100644
--- a/plugins/background/monitor-background.c
+++ b/plugins/background/monitor-background.c
@@ -63,6 +63,7 @@ build_monitor_background (MonitorBackground *mb)
 
     mb->window = gtk_window_new (GTK_WINDOW_TOPLEVEL);
     gtk_window_set_type_hint (GTK_WINDOW (mb->window), GDK_WINDOW_TYPE_HINT_DESKTOP);
+    gtk_window_set_decorated (GTK_WINDOW (mb->window), FALSE);
 
     // Set keep below so muffin recognizes the backgrounds and keeps them
     // at the bottom of the bottom window layer (under file managers, etc..)
